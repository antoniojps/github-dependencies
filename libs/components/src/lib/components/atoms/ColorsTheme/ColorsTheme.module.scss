.container {
  display: inline-flex;
  align-items: center;
}

.sample {
  display: block;
  width: var(--size-xs2);
  height: var(--size-xs2);
}
