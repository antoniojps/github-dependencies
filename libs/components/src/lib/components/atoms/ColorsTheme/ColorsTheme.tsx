import React, { useMemo } from 'react';
import styles from './ColorsTheme.module.scss';

type Props = {
  colors: string[] | readonly string[] | readonly (readonly string[])[];
};

export const ColorsTheme = ({ colors }: Props) => {
  const colorsComputed = useMemo(() => {
    const color = colors[6];
    if (typeof color !== 'string' && color instanceof Array) return color;
    return colors.slice(0, 6);
  }, [colors]);

  return (
    <div className={styles.container}>
      {colorsComputed.map((color) => (
        <span className={styles.sample} style={{ background: color }} />
      ))}
    </div>
  );
};
