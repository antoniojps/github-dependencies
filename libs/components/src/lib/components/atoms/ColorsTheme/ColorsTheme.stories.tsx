import React from 'react';
import { ColorsTheme } from './ColorsTheme';
import { colorSchemes } from '@nivo/colors';

export default {
  title: 'atoms/ColorsTheme',
};

export const Basic = () => <ColorsTheme colors={colorSchemes.nivo} />;
export const WithAlternative = () => <ColorsTheme colors={colorSchemes.set3} />;
export const WithSequential = () => <ColorsTheme colors={colorSchemes.blues} />;
