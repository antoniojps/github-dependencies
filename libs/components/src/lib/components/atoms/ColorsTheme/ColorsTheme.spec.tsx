import React from 'react';
import { render } from '@testing-library/react';
import { ColorsTheme } from './ColorsTheme';
import { colorSchemes } from '@nivo/colors';

describe('ColorsTheme', () => {
  it('renders with nivo (string[])', () => {
    const { asFragment } = render(<ColorsTheme colors={colorSchemes.nivo} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders with blues (readonly (readonly string[])[])', () => {
    const { asFragment } = render(<ColorsTheme colors={colorSchemes.blues} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders with set3 (readonly string[])', () => {
    const { asFragment } = render(<ColorsTheme colors={colorSchemes.set3} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
