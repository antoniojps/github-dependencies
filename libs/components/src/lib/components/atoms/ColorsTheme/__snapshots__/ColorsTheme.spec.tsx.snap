// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ColorsTheme renders with blues (readonly (readonly string[])[]) 1`] = `
<DocumentFragment>
  <div
    class="container"
  >
    <span
      class="sample"
      style="background: rgb(239, 243, 255);"
    />
    <span
      class="sample"
      style="background: rgb(198, 219, 239);"
    />
    <span
      class="sample"
      style="background: rgb(158, 202, 225);"
    />
    <span
      class="sample"
      style="background: rgb(107, 174, 214);"
    />
    <span
      class="sample"
      style="background: rgb(49, 130, 189);"
    />
    <span
      class="sample"
      style="background: rgb(8, 81, 156);"
    />
  </div>
</DocumentFragment>
`;

exports[`ColorsTheme renders with nivo (string[]) 1`] = `
<DocumentFragment>
  <div
    class="container"
  >
    <span
      class="sample"
      style="background: rgb(232, 193, 160);"
    />
    <span
      class="sample"
      style="background: rgb(244, 117, 96);"
    />
    <span
      class="sample"
      style="background: rgb(241, 225, 91);"
    />
    <span
      class="sample"
      style="background: rgb(232, 168, 56);"
    />
    <span
      class="sample"
      style="background: rgb(97, 205, 187);"
    />
    <span
      class="sample"
      style="background: rgb(151, 227, 213);"
    />
  </div>
</DocumentFragment>
`;

exports[`ColorsTheme renders with set3 (readonly string[]) 1`] = `
<DocumentFragment>
  <div
    class="container"
  >
    <span
      class="sample"
      style="background: rgb(141, 211, 199);"
    />
    <span
      class="sample"
      style="background: rgb(255, 255, 179);"
    />
    <span
      class="sample"
      style="background: rgb(190, 186, 218);"
    />
    <span
      class="sample"
      style="background: rgb(251, 128, 114);"
    />
    <span
      class="sample"
      style="background: rgb(128, 177, 211);"
    />
    <span
      class="sample"
      style="background: rgb(253, 180, 98);"
    />
  </div>
</DocumentFragment>
`;
